### PR TITLE
Add try-catch block around call to std::imbue()

### PR DESCRIPTION
**Pull request by tobim**

### CODE REVIEW

File: libvast/src/system/accountant.cpp [hunk 1]
```diff
@@ -109,7 +109,12 @@ void accountant_state::command_line_heartbeat() {
   if (logger && logger->verbosity() >= CAF_LOG_LEVEL_INFO
       && accumulator.node.events > 0) {
     std::ostringstream oss;
-    oss.imbue(std::locale(""));
+    try {
+      oss.imbue(std::locale(""));
+    } catch (const std::exception& e) {
+      VAST_DEBUG_ANON("accoutant failed to set locale the statistics output:",
```
Comment (Neverlord) [inline libvast/src/system/accountant.cpp]: Why are you using `VAST_DEBUG_ANON` here? `VAST_DEBUG(self, ...)` should be fine in this context. Also the sentence is off. Did you mean "... to set the locale for statistics ..."?

Comment (tobim) [reply libvast/src/system/accountant.cpp]: The state type didn't have a `self` member. That is rectified with the follow up commits.

File: libvast/src/system/accountant.cpp [hunk 2]
```diff
@@ -112,8 +112,8 @@ void accountant_state::command_line_heartbeat() {
     try {
       oss.imbue(std::locale(""));
     } catch (const std::exception& e) {
-      VAST_DEBUG_ANON("accoutant failed to set locale the statistics output:",
-                      e.what());
+      VAST_DEBUG(self,
```
Comment (mavam) [inline libvast/src/system/accountant.cpp]: Should this perhaps be a warning?

File: libvast/vast/system/accountant.hpp [hunk 3]
```diff
@@ -81,6 +70,20 @@ using accountant_type = caf::typed_actor<
   caf::reacts_to<telemetry_atom>>;
 // clang-format on
 
+/// @relates accountant
+struct accountant_state {
+  static constexpr const char* name = "accountant";
+  accountant_type::stateful_pointer<accountant_state> self;
```
Comment (Neverlord) [inline libvast/vast/system/accountant.hpp]: It seems like this pointer is never initialized. The state can have a constructor that takes the self pointer during construction.

File: libvast/src/system/accountant.cpp (hunk 2 above)
Comment (tobim) [reply libvast/src/system/accountant.cpp]: Imho warnings should be reserved for problems that affect the user in more serious way. For example if data could not be retrieved. I don't think the statistics formatting warrants that.

Comment (mavam) [reply libvast/src/system/accountant.cpp]: Good point.

File: libvast/vast/system/accountant.hpp (hunk 3 above)
Comment (tobim) [reply libvast/vast/system/accountant.hpp]: For some reason I thought this would work even without an explicitly defined constructor. Alas, it does not.

**Pull request merged**